prestodb/transaction.py

```diff
@@ -0,0 +1,81 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+
+from prestodb import constants
+import prestodb.exceptions
+
+
+logger = logging.getLogger(__name__)
+
+
+class Transaction(object):
+    def __init__(self, request):
+        self._request = request
+        self._id = 'NONE'
+
+    @property
+    def id(self):
+        return self._id
+
+    def begin(self):
+        response = self._request.post('START TRANSACTION')
+        if not response.ok:
+            raise prestodb.exceptions.DatabaseError(
+                'failed to start transaction: {}'.format(response.status_code))
+        transaction_id = response.headers.get(
```

Comment:
is `get()` going to return `None` if `headers` does not contain `HEADER_STARTED_TRANSACTION`?

Reply:
Interestingly no, because [`HTTPHeaderDict.__getitem__`](https://github.com/shazow/urllib3/blob/master/urllib3/_collections.py#L101) gets [`self._container[key.lower()]`](https://github.com/shazow/urllib3/blob/master/urllib3/_collections.py#L151) where `self._container` is an [`collections.OrderedDict`](https://github.com/shazow/urllib3/blob/master/urllib3/_collections.py#L137).

Good catch, thanks!

Reply:
Yes `import requests; requests.Response().headers.get('not there')` returns `None` and does not raise a `KeyError` exception.